src/Domain/Creators/Cylinder.cpp

```diff
@@ -76,13 +100,53 @@ Domain<3> Cylinder::create_domain() const noexcept {
       pairs_of_faces.push_back(std::move(south));
     }
   }
+
+  std::vector<DirectionMap<
+      3, std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>>>
+      boundary_conditions_all_blocks{};
+  if (boundary_condition_ != nullptr) {
+    // Note: The first block in each disk is the central cube.
+    boundary_conditions_all_blocks.resize((1 + 4 * number_of_shells) *
+                                          number_of_discs);
+
+    // Boundary conditions in z
+    for (size_t block_id = 0; not is_periodic_in_z_ and
+                              block_id < boundary_conditions_all_blocks.size();
+         ++block_id) {
+      if (block_id < (1 + number_of_shells * 4)) {
+        boundary_conditions_all_blocks[block_id][Direction<3>::lower_zeta()] =
+            boundary_condition_->get_clone();
+      }
+      if (block_id >=
+          boundary_conditions_all_blocks.size() - (1 + number_of_shells * 4)) {
+        boundary_conditions_all_blocks[block_id][Direction<3>::upper_zeta()] =
+            boundary_condition_->get_clone();
+      }
+    }
+    // Radial boundary conditions
+    ASSERT(radial_partitioning_.empty(),
+           "We currently do not support multiple radial partitionings.");
```

Comment:
Add something like "... with boundary conditions. Please add support if you need this feature."